builtin/logical/aws/path_config_root_test.go

```diff
@@ -324,10 +324,125 @@ func TestBackend_PathConfigRoot_STSFallback_defaultEndpointRegion(t *testing.T)
 	}
 	if len(cfgs) != 1 {
 		t.Fatalf("got %d configs, but expected 1", len(cfgs))
+	} else {
+		cfg := cfgs[0]
+		if *(cfg.Endpoint) != matchingSTSEndpoint(*(cfg.Region)) {
+			t.Fatalf("region and endpoint didn't match: %s vs. %s", *(cfg.Region), *(cfg.Endpoint))
+		}
 	}
 }
 
-// TestBackend_PathConfigRoot_IAM_defaultEndpointRegion ensures taht if no endpoints are specified, we can still
+// TestBackend_PathConfigRoot_IAM_specifiedRegion ensures that if a region is set, we get a good config (with a blank
+// endpoint)
+func TestBackend_PathConfigRoot_IAM_specifiedRegion(t *testing.T) {
+	config := logical.TestBackendConfig()
+	config.StorageView = &logical.InmemStorage{}
+	config.System = &testSystemView{}
+
+	b := Backend(config)
+	if err := b.Setup(context.Background(), config); err != nil {
+		t.Fatal(err)
+	}
+
+	desiredRegion := "us-west-2"
+
+	configData := map[string]interface{}{
+		"access_key":              "AKIAEXAMPLE",
+		"secret_key":              "RandomData",
+		"max_retries":             10,
+		"username_template":       defaultUserNameTemplate,
+		"region":                  desiredRegion,
+		"role_arn":                "",
+		"identity_token_audience": "",
+		"identity_token_ttl":      int64(0),
+	}
+
+	configReq := &logical.Request{
+		Operation: logical.UpdateOperation,
+		Storage:   config.StorageView,
+		Path:      "config/root",
+		Data:      configData,
+	}
+
+	_, err := b.HandleRequest(context.Background(), configReq)
+	if err != nil {
+		t.Fatalf("bad: config writing failed: err: %v", err)
+	}
+
+	cfgs, err := b.getRootConfigs(context.Background(), config.StorageView, "iam", b.Logger())
+	if err != nil {
+		t.Fatalf("couldn't get IAM configs with default region/endpoints: %v", err)
+	}
+
+	if len(cfgs) != 1 {
+		t.Fatalf("got %d configs, but expected 1", len(cfgs))
+	}
+
+	if *(cfgs[0].Endpoint) != "" {
```

Comment:
The reason why we need the endpoint to be held blank is that in official AWS IAM, the endpoint being set seems to fail the client setup, even if the endpoint matches what is specified on the AWS Endpoints list. Apparently it _needs_ to be blank (or some value I haven't been able to find.)

(see: https://docs.aws.amazon.com/general/latest/gr/iam-service.html)